src/Ionide.ProjInfo.ProjectSystem/Environment.fs

```diff
@@ -122,32 +124,25 @@ module Environment =
                 |> Option.bind
                     (fun d ->
                         let di = DirectoryInfo d
-                        if di.Exists then Some di else None)
 
-             defaultArg fromEnv FSIRefs.defaultDotNetSDKRoot)
-
-    let private maxVersionWithThreshold (minVersion: FSIRefs.NugetVersion) (versions: FSIRefs.NugetVersion []) =
-        versions
-        |> Array.filter (fun v -> FSIRefs.compareNugetVersion v minVersion >= 0) // get all versions that compare as greater than the minVersion
-        |> Array.sortWith FSIRefs.compareNugetVersion
-        |> Array.tryLast
+                        if di.Exists then
+                            Some di
+                        else
+                            None)
 
-    /// because 3.x is the minimum SDK that we support for FSI, we want to float to the latest
-    /// 3.x sdk that the user has installed, to prevent hard-coding.
-    let latest3xSdkVersion (dotnetRoot: DirectoryInfo) =
```

Comment:
I removed these two functions because they were too specialized. by exposing the core function callers can express the version ranges they need to without changes here.